File: Ejercicio 012/banco.java
class Cuenta {
    String numero;
    String titular;
    double saldo;
    String estado;

    public Cuenta(String numero, String titular, double saldo, String estado) {
        this.numero = numero;
        this.titular = titular;
        this.saldo = saldo;
        this.estado = estado;
    }

    public void mostrarInfo() {
        System.out.println("Cuenta: " + numero);
        System.out.println("Titular: " + titular);
        System.out.println("Saldo: " + saldo);
        System.out.println("Estado: " + estado);
    }

    public void operar() {}
}


class CuentaAhorros extends Cuenta {
    double interes = 0.02;

    public CuentaAhorros(String numero, String titular, double saldo, String estado) {
        super(numero, titular, saldo, estado);
    }

    public void operar() {
        double ganancia = saldo * interes;
        System.out.println("Interés generado: " + ganancia);
    }
}


class CuentaCorriente extends Cuenta {
    double limiteSobregiro = 500000;

    public CuentaCorriente(String numero, String titular, double saldo, String estado) {
        super(numero, titular, saldo, estado);
    }

    public void operar() {
        System.out.println("Tiene sobregiro disponible de: " + limiteSobregiro);
    }
}


class CuentaEstudiante extends Cuenta {

    public CuentaEstudiante(String numero, String titular, double saldo, String estado) {
        super(numero, titular, saldo, estado);
    }

    public void operar() {
        System.out.println("Cuenta sin comisiones");
    }
}


public class Main {
    public static void main(String[] args) {

        Cuenta c1 = new CuentaAhorros("001", "Ana", 1000000, "Activa");
        Cuenta c2 = new CuentaCorriente("002", "Carlos", 500000, "Activa");
        Cuenta c3 = new CuentaEstudiante("003", "Luis", 200000, "Activa");

        Cuenta[] cuentas = {c1, c2, c3};

        for (Cuenta c : cuentas) {
            System.out.println("Tipo: " + c.getClass().getSimpleName());
            c.mostrarInfo();
            c.operar();
            System.out.println("-------------------");
        }
    }
}

File: Ejercicio 012/banco.py
class Cuenta:
    def __init__(self, numero, titular, saldo, estado):
        self.numero = numero
        self.titular = titular
        self.saldo = saldo
        self.estado = estado

    def mostrar_info(self):
        print("Cuenta:", self.numero)
        print("Titular:", self.titular)
        print("Saldo:", self.saldo)
        print("Estado:", self.estado)

    def operar(self):
        pass


class CuentaAhorros(Cuenta):
    def operar(self):
        interes = self.saldo * 0.02
        print("Interés generado:", interes)


class CuentaCorriente(Cuenta):
    def operar(self):
        print("Tiene sobregiro disponible de: 500000")


class CuentaEstudiante(Cuenta):
    def operar(self):
        print("Cuenta sin comisiones")



c1 = CuentaAhorros("001", "Ana", 1000000, "Activa")
c2 = CuentaCorriente("002", "Carlos", 500000, "Activa")
c3 = CuentaEstudiante("003", "Luis", 200000, "Activa")

cuentas = [c1, c2, c3]

for c in cuentas:
    print("Tipo:", type(c).__name__)
    c.mostrar_info()
    c.operar()
    print("-------------------")
